package.json

```diff
@@ -83,7 +84,8 @@
     "lodash.camelcase": "^4.3.0",
     "lodash.kebabcase": "^4.1.1",
     "requireindex": "^1.2.0",
-    "snake-case": "^3.0.3"
+    "snake-case": "^3.0.3",
+    "typescript": "^5.2.2"
```

Comment:
Are you sure we need to add `typescript` as a dependency? Can you explain the justification for it? We should be careful about adding large dependencies especially when users may be on different versions of it (just like users may be on different versions of Ember itself).

Reply:
Yes, it's required to set it's ts.sys to handle gts in a way that supports type aware lint. But we could set it as peer dependency.

Reply:
That's an interesting idea. If we make it an optional peer dependency, then users who don't want or need typescript can avoid it right? And can you make your code resilient in the event that typescript is not installed? Seems like we ignore it for users who don't have it.

Reply:
And it's only required for type aware actually.

Reply:
@patricklx, @bmish also because we are relying on monkey patching `ts` even when `typescript-eslint` is running we probably we need to make sure `typescript-eslint` and `eslint-plugin-ember` are using the same instance of typescript when used by users. I guess making `typescript` as well as `typescript-eslint` peerDeps can help with this too?

Reply:
now as optional peer dep

Reply:
Yes, I think i missed a commit during rebase...  :(